src/EnergyPlus/OutputReports.cc

```diff
@@ -1572,35 +1567,41 @@ void DetailsForSurfaces(EnergyPlusData &state, int const RptType) // (1=Vertices
                            << format("{:.2R}", state.dataSurface->Surface(surf).Height) << ","
                            << format("{:.2R}", state.dataSurface->Surface(surf).Reveal) << ",";
                 if (state.dataSurface->SurfIntConvCoeffIndex(surf) > 0) {
-                    {
-                        auto const SELECT_CASE_var(
-                            state.dataSurface->UserIntConvectionCoeffs(state.dataSurface->SurfIntConvCoeffIndex(surf)).OverrideType);
-                        if (SELECT_CASE_var == ConvCoefValue) {
-                            IntConvCoeffCalc = "User Supplied Value";
-                        } else if (SELECT_CASE_var == ConvCoefSchedule) {
-                            IntConvCoeffCalc = "User Supplied Schedule";
-                        } else if (SELECT_CASE_var == ConvCoefUserCurve) {
-                            ExtConvCoeffCalc = "User Supplied Curve";
-                        } else if (SELECT_CASE_var == ConvCoefSpecifiedModel) {
-                            ExtConvCoeffCalc = "User Specified Model";
-                        }
+                    switch (state.dataSurface->UserIntConvectionCoeffs(state.dataSurface->SurfIntConvCoeffIndex(surf)).OverrideType) {
+                    case ConvectionConstants::ConvCoefOverrideType::Value:
```

Comment:
This could be a lookup in a constexpr `std::array<std::string_view, ConvCoefOverrideType::Num>`.

Reply:
Sounds good, I'll make it happen.

Reply:
This may be more of a stylistic question rather than a functionality/efficiency question. This particular constexpr string_view array that you're suggesting is only used at one spot (inside of this IF statement). Should it stay there inside of the IF statement, or should it be hauled up to the top of the namespace or dropped in the header? I'm guessing that since it's a compile-time constant, it doesn't matter if it is hit every time we drop into this IF block, but maybe I'm wrong on that.

Reply:
You're correct, there's no runtime cost regardless of where you put it. You can leave it in place for readability if that is the only place it is used.